Kickstarter-iOS/Features/PaginationExample/PaginationExampleView.swift

```diff
@@ -0,0 +1,82 @@
+import KsApi
+import SwiftUI
+
+private struct PaginationExampleProjectCell: View {
+  let title: String
+  var body: some View {
+    Text(title)
+      .padding(.all, 10)
+  }
+}
+
+private struct PaginationExampleProjectList: View {
+  @Binding var projectIdsAndTitles: [(Int, String)]
+  @Binding var showProgressView: Bool
+  @Binding var statusText: String
+
+  let onRefresh: @Sendable() -> Void
+  let onDidShowProgressView: @Sendable() -> Void
+
+  var body: some View {
+    HStack {
+      Spacer()
+      Text("👉 \(statusText)")
+      Spacer()
+    }
+    .padding(EdgeInsets(top: 20, leading: 0, bottom: 20, trailing: 0))
+    .background(Color.yellow)
+    List {
+      ForEach(projectIdsAndTitles, id: \.0) {
+        let title = $0.1
+        PaginationExampleProjectCell(title: $0.1)
+      }
+      if showProgressView {
+        HStack {
+          Spacer()
+          Text("Loading 😉")
+            .onAppear {
+              onDidShowProgressView()
+            }
+          Spacer()
+        }
+        .background(Color.yellow)
+      }
+    }
+    .refreshable {
+      onRefresh()
+    }
+  }
+}
+
+public struct PaginationExampleView: View {
+  @StateObject private var viewModel = PaginationExampleViewModel()
+
+  public var body: some View {
+    let capturedViewModel = viewModel
```

Comment:
Could you explain why we need to capture the viewModel here instead of using the @StateObject var directly in ` PaginationExampleProjectList`?

Reply:
It's a side-effect of these two closures being marked `@Sendable`. Without the capture, you get useful, but not terribly clear error:

```Main actor-isolated property 'viewModel' can not be referenced from a Sendable closure```

Reply:
Fixed, see note above about removing `@Sendable`